src/main/java/core/basesyntax/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,207 @@
+                drivers.add(makeDriver(driverResultSet));
+            }
+            car.setDrivers(drivers);
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to GET car from DB, id=" + id, e);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllCarsString = "SELECT c.car_id, car_model AS model, c.manufacturer_id, "
+                + "manufacturer_name AS made_by, manufacturer_country AS country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m "
+                + "ON c.manufacturer_id=m.manufacturer_id "
+                + "WHERE c.deleted=false;";
+        String getCarDrivers = "SELECT cd.driver_id, d.driver_name AS driver_name, "
+                + "d.driver_license AS license "
+                + "FROM cars_drivers cd "
+                + "INNER JOIN drivers d "
+                + "ON cd.driver_id=d.driver_id "
+                + "WHERE car_id=? AND d.deleted=false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement carsStatement = connection.prepareStatement(getAllCarsString);
+                PreparedStatement driversStatement = connection.prepareStatement(getCarDrivers)) {
+            ResultSet carsResultSet = carsStatement.executeQuery();
+            while (carsResultSet.next()) {
+                Car car = makeCar(carsResultSet);
+                List<Driver> drivers = new ArrayList<>();
+                driversStatement.setLong(1, car.getId());
+                ResultSet driverResultSet = driversStatement.executeQuery();
+                while (driverResultSet.next()) {
+                    drivers.add(makeDriver(driverResultSet));
+                }
+                car.setDrivers(drivers);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to GET ALL cars from DB", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarString = "UPDATE cars SET car_id=?, manufacturer_id=?, car_model=?"
+                + " WHERE car_id=? AND deleted=false;";
+        String deleteLinksString = "DELETE FROM cars_drivers WHERE car_id=?;";
+        String newLinksString = "INSERT INTO cars_drivers (car_id, driver_id"
+                + ") VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement carUpdateStatement = connection.prepareStatement(updateCarString);
+                PreparedStatement deleteLinksStatement = connection
+                        .prepareStatement(deleteLinksString);
+                PreparedStatement newLinksStatement = connection.prepareStatement(newLinksString)) {
+            carUpdateStatement.setLong(1, car.getId());
```

Comment:
```suggestion
            carUpdateStatement.setLong(1, car.getId());
```
We are updating by Id. The id by itself won't be changed

Reply:
done